python/whylogs/api/writer/whylabs.py

```diff
@@ -132,22 +155,40 @@ def _upload_whylabs(
                 "variable or on your write method"
             )
 
-        upload_url = upload_url or self._get_upload_url(dataset_timestamp=dataset_timestamp)
+        upload_url = upload_url or self._get_upload_url(
+            dataset_timestamp=dataset_timestamp, reference=reference, reference_alias=reference_alias
+        )
         api_key_id = self._api_key[:10] if self._api_key else None
-        try:
-            with open(profile_path, "rb") as f:
-                http_response = requests.put(upload_url, data=f.read())
-                if http_response.status_code == 200:
-                    logger.info(
-                        f"Done uploading {self._org_id}/{self._dataset_id}/{dataset_timestamp} to "
-                        f"{self.whylabs_api_endpoint} with API token ID: {api_key_id}"
-                    )
-                return http_response
-        except requests.RequestException as e:
-            logger.info(
-                f"Failed to upload {self._org_id}/{self._dataset_id}/{dataset_timestamp} to "
-                + f"{self.whylabs_api_endpoint} with API token ID: {api_key_id}. Error occurred: {e}"
-            )
+        if not reference:
+            try:
+                with open(profile_path, "rb") as f:
+                    http_response = requests.put(upload_url, data=f.read())
+                    if http_response.status_code == 200:
+                        logger.info(
+                            f"Done uploading {self._org_id}/{self._dataset_id}/{dataset_timestamp} to "
+                            f"{self.whylabs_api_endpoint} with API token ID: {api_key_id}"
```

Comment:
Hopefully the API token is not secret :)

Reply:
I got scared for a bit, but if you see `api_key_id`, it's already only the first 10 characters of the complete key!
So I think it's good.